meta-mender-core/classes/mender-setup.bbclass

```diff
@@ -176,6 +176,67 @@ python() {
         bb.fatal("MENDER_PARTITION_ALIGNMENT_MB is deprecated. Please define MENDER_PARTITION_ALIGNMENT_KB instead.")
 }
 
+addhandler mender_vars_handler
+python mender_vars_handler() {
+    if(isinstance(e, bb.event.ParseCompleted)):
+        from bb import data
+        import os
+        import re 
+        import json
+
+        path = d.getVar("LAYERDIR_MENDER")
+        path = os.path.join(path, "conf/mender-vars.json")
+
+        if( (os.path.isfile(path))):
+            mender_vars = {}
+            with open(path, "r") as f:
+                mender_vars = json.load(f)
+
+            for k in d.keys():
+                if k.startswith("MENDER_"):
+                    if ( k not in mender_vars.keys() ):
+                    # Warn if user has defined some new (unused) MENDER_.* variables
+                        bb.warn("\"%s\" is not a real MENDER_ variable" % k)
```

Comment:
I suggest using the message: `"\"%s\" is not a recognized MENDER_ variable. Typo?" % k`